### ISSUE TITLE
auto_link on Windows (MSVC)

### ISSUE THREAD

**Issue by ztl8702** [nelhage/rules_boost#112]
On Windows, with MSVC compiler, the auto_link macros of Boost generate such directives:

```
#pragma comment(lib, "libboost_thread-vc140-mt-x64-1_67.lib")
```

into header files, such as `boost/thread.hpp`.

However, Bazel will build `@boost//:thread` into `bazel-bin\external\boost\thread.lib` (rather than `bazel-bin\external\boost\libboost_thread-vc140-mt-x64-1_67.lib`)

When I build a cc_binary that depends on, say `@boost//:thread`, it will fail to link:

```
LINK : fatal error LNK1104: cannot open file 'libboost_thread-vc140-mt-x64-1_67.lib'
Target //test-ui-app:win32 failed to build
INFO: Elapsed time: 2.819s, Critical Path: 0.20s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

because that `#pragma comment` directive will cause `link.exe` to search for the non-existent `libboost_thread-vc140-mt-x64-1_67.lib`, even though Bazel will also pass the valid path of `thread.lib` to link.exe.

So, can we disable Boost auto_link on Windows, or workaround this somehow?

**Comment by ztl8702** [comment]
Answering my own question: I can specify `defines = ["BOOST_ALL_NO_LIB"]` in my cc_binary.

However, should this be an option in rules_boost?

**Comment by ztl8702** [comment]
Actually, AFAIK, BOOST_ALL_NO_LIB need to be set in rules_boost targets, otherwise the intermediate .lib / .obj files will still contain the autolink #pragma directives

**Comment by nelhage** [comment]
I'd be open to a PR for this. I don't have a Windows test environment and have very minimal familiarity with Windows toolchains and builds so I'm not sure I can comment directly.

**Comment by ztl8702** [comment]
> I'd be open to a PR for this. I don't have a Windows test environment and have very minimal familiarity with Windows toolchains and builds so I'm not sure I can comment directly.

Yeah I will file a PR if I am ready (currently still ramping up with Bazel concepts)

**Comment by ztl8702** [comment]
Fixed in #123 thanks to @Vertexwahn
